.travis.yml

```diff
@@ -5,12 +19,17 @@ rvm:
 - 2.0.0
 
 cache:
-  directories:
-  - $HOME/gcloud/
-  - vendor/bundle
+  - directories:
+    - $HOME/gcloud/
+    - vendor/bundle
 
 env:
-- PATH=$PATH:$HOME/gcloud/google-cloud-sdk/bin GOOGLE_APPLICATION_CREDENTIALS=$TRAVIS_BUILD_DIR/client_secrets.json #Other environment variables on same line
+- TEST_DIR=storage
+  PATH=$PATH:$HOME/gcloud/google-cloud-sdk/bin
+  GOOGLE_APPLICATION_CREDENTIALS=$TRAVIS_BUILD_DIR/client_secrets.json
+- TEST_DIR=pubsub
+  PATH=$PATH:$HOME/gcloud/google-cloud-sdk/bin
+  GOOGLE_APPLICATION_CREDENTIALS=$TRAVIS_BUILD_DIR/client_secrets.json
```

Comment:
This is interesting to use the build matrix for each sample's test suite. Wouldn't it be simpler to run them sequentially using one build?